### DISCUSSION TITLE
Is it possible to point the logo to a path for random output?

### DISCUSSION THREAD

**Discussion by ayala** [fastfetch-cli/fastfetch#698] [General]
I think I've seen this functionality but don't see anything in the documentation, also not sure it it was a script or natively supported.

Is there a way to point ff to folder of logos and have it randomly use different .png each time?

TIA!

**Comment by CarterLi** [comment]
```bash
fastfetch --iterm $(ls /path/to/images/*.png | shuf -n 1)
```

**Reply by ayala** [reply]
Thanks but I'm having trouble implementing this. I'm on Mac with iTerm. the command plus the piped shut portion work well enough when run by itself but I can't get it to work with "--iterm" in terminal or in my config, which doesn't make sense.Then again I;m pretty noob.

**Reply by CarterLi** [reply]
> but I can't get it to work with "--iterm" in terminal or in my config

It uses bash syntax. It works only in bash compatible shell, but not in fastfetch config file.

**Reply by ayala** [reply]
yeah been learning all day lol, thanks!

**Comment by CarterLi** [comment]
A new release was cut to permit this syntax in jsonc config file

**Reply by ayala** [reply]
This is so awesome. thank you I finally got it to work. 

Solution is as you described in your changelog. `"source": "$(ls ~/.config/fastfetch/logos/*.png | shuf -n 1)"`

**Comment by younsl** [comment]
When you install Fastfetch for the first time, you won't get a configuration file. So the first step is to generate a simple config file. For this, run the command:

```bash
fastfetch --gen-config
```

By default, a configuration file is created in the `$HOME/.config/fastfetch/config.json` path.

randomly displays one of all .png files in `~/.config/fastfetch/images/`.

Example fastfetch config:

```json
{
  "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
  "logo": {
    "type": "auto",
    "source": "~/.config/fastfetch/images/*.png"
  },
  "modules": [
    "title",
    // ... omitted for brevity ...
    "colors"
  ]
}
```

Only executes `fastfetch` command each time the terminal is launched. The necessary options have already been declared in the config file, so no additional flags are needed.

```bash
echo "fastfetch" >> $HOME/.zshrc
```

**Reply by CarterLi** [reply]
`"source": "~/.config/fastfetch/images/*.png"` is enough

**Reply by younsl** [reply]
@CarterLi Oh, there was a more concise way to set it up. I reflected this in my comment.

**Reply by Panelgamer** [reply]
> `"source": "~/.config/fastfetch/images/*.png"` is enough

And, if i want to add .jpg? and, if i want to crop the image size?, i mean, the images just crashed (become bigger than the terminal itself and dont show up).
